### PR TITLE
executor: Fix replace statement conflict with multiple row bug.

### PR DESCRIPTION
**Pull request by shenli**
Fix: https://github.com/pingcap/tidb/issues/1012

@coocood @zimulala

### CODE REVIEW

File: executor/executor_write.go [hunk 1]
```diff
@@ -818,54 +818,58 @@ func (e *ReplaceExec) Next() (*Row, error) {
...
-			changeHandle := false
-			oldRow, err1 := e.Table.Row(e.ctx, h)
-			if err1 != nil {
-				return nil, errors.Trace(err1)
+		oldRow, err1 := e.Table.Row(e.ctx, h)
+		if err1 != nil {
+			return nil, errors.Trace(err1)
+		}
+		// Check if need to replace the current row.
+		needToReplace := false
```
Comment (coocood) [inline executor/executor_write.go]: When 'needToReplace` will be false?

File: executor/executor_write.go [hunk 2]
```diff
@@ -818,54 +818,58 @@ func (e *ReplaceExec) Next() (*Row, error) {
...
-				if err != nil {
-					return nil, errors.Trace(err)
-				}
-				_, err = e.Table.AddRecord(e.ctx, row)
-				variable.GetSessionVars(e.ctx).AddAffectedRows(1)
-				continue
-			}
 		}
-		// Replace the current row data.
-		if err = e.replaceRow(h, row); err != nil {
```
Comment (coocood) [inline executor/executor_write.go]: This method can be deleted.

Comment (shenli) [inline executor/executor_write.go]: yes

**Comment by shenli** [comment]
PTAL @coocood @zimulala

File: util/types/datum.go [hunk 3]
```diff
@@ -1247,3 +1247,26 @@ func MinNotNullDatum() Datum {
 func MaxValueDatum() Datum {
 	return Datum{k: KindMaxValue}
 }
+
+// SameDatums compare if a and b contains the same datum values.
+func SameDatums(a []Datum, b []Datum) (bool, error) {
```
Comment (coocood) [inline util/types/datum.go]: Rename to EqualDatumns.

**Comment by shenli** [comment]
PTAL @coocood

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**